pro/src/data_access/RandomRecipeDataAccessObject.java

```diff
@@ -0,0 +1,89 @@
+package data_access;
+
+import entity.Recipe;
+import entity.RecipeFactory;
+import entity.User;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.json.JSONObject;
+import service.recommendation.use_case.RecommendationDataAccessInterface;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RandomRecipeDataAccessObject implements RecommendationDataAccessInterface {
+    private final RecipeFactory recipeFactory;
+
+    public RandomRecipeDataAccessObject(RecipeFactory recipeFactory) {
+        this.recipeFactory = recipeFactory;
+    }
+
+    private JSONObject random() {
+        OkHttpClient client = new OkHttpClient();
+        String link = "https://www.themealdb.com/api/json/v1/1/random.php";
+
+        try {
+            Request request = new Request.Builder().url(link).get().build();
+            Response response = client.newCall(request).execute();
+            JSONObject responseBody = new JSONObject(response.body().string());
+            return responseBody;
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Recipe convertIntoRecipe(JSONObject recipeJson) {
```

Comment:
To avoid duplicate code, you can inject an instance attribute called ApiRecipeDataAccessObject and call the method convertIntoRecipe by using the adapter design pattern.